cmd/server/cmd/root.go

```diff
@@ -102,25 +99,47 @@ func Execute() {
 	// Authenticator + ATLS creds
 	opts := []grpc.ServerOption{
 		// grpc.ChainUnaryInterceptor(auth.AuthUnaryInterceptor),
-		grpc.ChainStreamInterceptor(auth.AuthStreamInterceptor, grpc_middleware.ChainStreamServer(
+		grpc.ChainStreamInterceptor(servAuth.AuthStreamInterceptor, grpc_middleware.ChainStreamServer(
 			grpc_zap.StreamServerInterceptor(logger.ZapInterceptor()))),
 		grpc.Creds(credentials.NewServerTLSFromCert(&cert)),
+		grpc.MaxConcurrentStreams(1),
 	}
-	grpcServer := grpc.NewServer(opts...)
+	grpcServer = grpc.NewServer(opts...)
 
+	transcribeServer = &TranscribeServer{
+		DBClient: dbclient,
+		auth:     *servAuth,
+		Store:    fsstore,
+	}
 	// Register STT Server with Transcribe server instance
 	sttservice.RegisterSttServiceServer(grpcServer,
-		&TranscribeServer{
-			DBClient: dbclient,
-			auth:     *servAuth,
-			Store:    fsstore,
-		})
+		transcribeServer)
 
 	// Run http and grpc server
-	for {
-		log.Fatal(grpcServer.Serve(tcpListen))
-		// log.Fatal(server.ListenAndServe())
-	}
+	go gracefullShutdown()
+
+	log.Fatal(grpcServer.Serve(tcpListen))
+
+	// Waiting for channel to get closed after graceful shutdown
+	<-idleConnectionsClosed
+
+}
+
+// Implements graceful shutdown
+func gracefullShutdown() {
+	sigint := make(chan os.Signal, 1)
+	signal.Notify(sigint, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+	<-sigint
+	log.Println("Shutting down server")
+
+	// Closing connection to Database and blob store
+	transcribeServer.DBClient.Close()
+	transcribeServer.Store.Close()
+	grpcServer.GracefulStop()
```

Comment:
```suggestion
	grpcServer.GracefulStop()
	transcribeServer.Store.Close()
	transcribeServer.DBClient.Close()
```
Сначала следует погасить grpc-сервер, чтобы он более не принимал соединений и не пытался писать в сторадж. Потом погасить сторадж и только после этого закрыть соединение с БД. Иначе есть большой риск словить на шатдауне панику.